packages/swapper/src/utils.ts

```diff
@@ -302,8 +301,11 @@ export const getInputOutputRate = ({
   return bn(buyAmountCryptoHuman).div(sellAmountCryptoHuman).toFixed()
 }
 
-export const isExecutableTradeQuote = (quote: TradeQuote): quote is ExecutableTradeQuote =>
-  !!quote.receiveAddress
+export const isExecutableTradeQuote = (quote: TradeQuote | TradeRate): quote is TradeQuote => {
+  if (quote.steps[0]!.source.includes(SwapperName.LIFI)) return !!quote.receiveAddress
+
+  return quote.quoteOrRate === 'quote'
+}
```

Comment:
re: first q, that would make a lot of sense actually, though not sure it makes sense *currently* as we are in the process of finding a way to make Li.Fi quotes/rate compliant and fetch final quotes using stepTransaction. Returning a `quote`, even though it's what we do, really doesn't fit the pattern
re: second one, actually cannot given our flow, since things referentially invalidate on wallety deps change for quotes to refetch, and you cannot continue to preview trade screen without a wallet connected

Reply:
Not super sure I understand this guy. My understanding is that anything with a `quoteOrRate` of `quote` will have a `receiveAddress`, and so the `return quote.quoteOrRate === 'quote'` part should be all that's necessary?

Reply:
@0xApotheosis FYI, this change is gone as Li.Fi now follows the rates then quotes standard flow following 8309

Previously (i.e before I rebased this this mo') however, that wasn't the case, and `quoteOrRate` didn't cut it for Li.Fi:
Yes, `quoteOrRate` is indeed the new discriminator as part of this PR, but we previously never wanted to refetch a final quote for Li.Fi, meaning that a Li.Fi quote was fetched at rate time as a *rate* and should stay as such, but was actually executable, which is why there was a special case for Li.Fi.

Now that it follows our usual pattern, no need for it anymore!

Reply:
44e0301847 and retested Li.Fi execution for sanity's sake:

https://jam.dev/c/6ae57343-9a9f-4ff3-a7ec-776ba8a9f63d